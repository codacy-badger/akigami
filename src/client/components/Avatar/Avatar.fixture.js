import random from 'lodash/random';
import Avatar from './Avatar';

const user = () => ({
  avatar: `https://randomuser.me/api/portraits/men/${random(99)}.jpg`,
});

export default [
  {
    component: Avatar,
    name: 'small',
    props: {
      src: user().avatar,
      size: 24,
    },
  },
  {
    component: Avatar,
    name: 'normal',
    props: {
      src: user().avatar,
      size: 40,
    },
  },
  {
    component: Avatar,
    name: 'large',
    props: {
      src: user().avatar,
      size: 100,
    },
  },
];
