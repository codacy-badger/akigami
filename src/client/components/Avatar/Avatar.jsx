import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

class Avatar extends PureComponent {
    static defaultProps = {
        src: null,
        size: 40,
        href: null,
        online: null,
        onClick: null,
        className: null,
    }
    static propTypes = {
        src: PropTypes.string,
        size: PropTypes.oneOfType([PropTypes.number, PropTypes.oneOf(['auto'])]),
        href: PropTypes.string,
        online: PropTypes.bool,
        onClick: PropTypes.func,
        className: PropTypes.string,
    }

    render() {
        const { src, href, onClick, size, online, className } = this.props;
        const props = {};
        let Tag = 'div';
        if (href) {
            Tag = 'a';
            props.href = href;
        }
        if (onClick) {
            Tag = 'button';
            props.onClick = onClick;
        }
        const onlineTarget = typeof online === 'object';
        const style = {
            backgroundImage: `url(${src})`,
            width: size,
            height: size,
        };
        if (size === 'auto') {
            style.width = '100%';
            style.height = 0;
            style.paddingBottom = '100%';
        }
        return (
            <Tag
                {...props}
                className={cx({
                    avatar: true,
                    offline: !onlineTarget && !online,
                    online: !onlineTarget && online,
                    [className]: className,
                })}
                style={style}
            />
        );
    }
}

export default Avatar;
