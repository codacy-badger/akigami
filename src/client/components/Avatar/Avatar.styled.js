import styled from 'emotion/react';

export const Image = styled('div')`
    background-size: cover;
    background-position: center;
    background-color: #424242;
    border-radius: 100px;
    width: ${({ size }) => size};
    height: ${({ size }) => size};
    background-image: url(${({ src }) => src});
`;

export default Image;
