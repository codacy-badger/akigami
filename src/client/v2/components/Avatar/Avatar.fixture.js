import Avatar from './Avatar';

export default [
  {
    component: Avatar,
    name: 'small',
    props: {
      src: 'https://randomuser.me/api/portraits/men/1.jpg',
      size: 24,
    },
  },
  {
    component: Avatar,
    name: 'normal',
    props: {
      src: 'https://randomuser.me/api/portraits/men/1.jpg',
      size: 40,
    },
  },
  {
    component: Avatar,
    name: 'large',
    props: {
      src: 'https://randomuser.me/api/portraits/men/1.jpg',
      size: 100,
    },
  },
];
