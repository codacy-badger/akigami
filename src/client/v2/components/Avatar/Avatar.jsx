import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Image } from './Avatar.styled';

class Avatar extends PureComponent {
  static propTypes = {
    src: PropTypes.string,
    size: PropTypes.number,
    className: PropTypes.string,
  };
  static defaultProps = {
    src: '',
    size: 40,
    className: '',
  };
  render() {
    const { src, size, className } = this.props;
    return <Image size={size} src={src} className={className} />;
  }
}

export default Avatar;
