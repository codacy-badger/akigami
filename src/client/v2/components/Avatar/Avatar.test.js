import { create as renderer } from 'react-test-renderer';
import createTestContext from 'react-cosmos-test/generic';
import Avatar from './Avatar.fixture';

Avatar.forEach((fixture) => {
  const res = createTestContext({ renderer, fixture });
  beforeEach(res.mount);
  test(`Avatar ${fixture.name} snapshot`, () => {
    expect(res.getWrapper().toJSON()).toMatchSnapshot();
  });
});

