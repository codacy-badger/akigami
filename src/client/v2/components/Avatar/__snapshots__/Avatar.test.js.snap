// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar large snapshot 1`] = `
<div
  className="css-6qe8gp"
  size={100}
  src="https://randomuser.me/api/portraits/men/1.jpg"
/>
`;

exports[`Avatar normal snapshot 1`] = `
<div
  className="css-6qe8gp"
  size={100}
  src="https://randomuser.me/api/portraits/men/1.jpg"
/>
`;

exports[`Avatar small snapshot 1`] = `
<div
  className="css-6qe8gp"
  size={100}
  src="https://randomuser.me/api/portraits/men/1.jpg"
/>
`;
